File: src/main/java/org/dxworks/githubminer/http/GithubHttpResponse.java
package org.dxworks.githubminer.http;

import lombok.Getter;
import org.dxworks.githubminer.pagination.PageLinks;
import org.dxworks.utils.java.rest.client.response.HttpResponse;

@Getter
public class GithubHttpResponse extends HttpResponse {

    private PageLinks pageLinks;

    public GithubHttpResponse(HttpResponse response) {
        super(response.getResponse());
        this.pageLinks = new PageLinks(this);
    }
}

File: src/main/kotlin/org/dxworks/githubminer/http/GithubHttpResponse.kt
package org.dxworks.githubminer.http

import org.dxworks.githubminer.dto.rate.RateLimit
import org.dxworks.githubminer.pagination.PageLinks
import org.dxworks.utils.java.rest.client.response.HttpResponse

class GithubHttpResponse(response: HttpResponse) : HttpResponse(response.response) {
    val pageLinks: PageLinks = PageLinks(this)
    val rateLimit: RateLimit = RateLimit(this)
}
